torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -257,3 +263,54 @@ def lexicon_decoder(
         sil_token=sil_token,
         unk_word=unk_word,
     )
+
+
+def _get_filenames(model: str) -> _PretrainedFiles:
+    MODEL_FILE_MAP = {
+        "librispeech": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm=None),
+        "librispeech-3-gram": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm="lm.bin"),
+        "librispeech-4-gram": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm="lm.bin"),
+    }
+
+    files = MODEL_FILE_MAP[model]
+    prefix = f"decoder-assets/{model}"
+    return _PretrainedFiles(
+        lexicon=f"{prefix}/{files.lexicon}",
+        tokens=f"{prefix}/{files.tokens}",
+        lm=f"{prefix}/{files.lm}" if files.lm is not None else None,
+    )
```

Comment:
Since the values of MODEL_FILE_MAP are identical, this function can be reduced to 

```python
def _get_filenames(model: str) -> _PretrainedFiles:
    prefix = f"decoder-assets/{model}"
    return _PretrainedFiles(
         lexicon=f"{prefix}/lexicon.txt",
         tokens=f"{prefix}/tokens.txt",
         lm=None if model is None else f"{prefix}/lm.bin",
)
```

note: The whole purpose of this helper function is to decouple the responsibility of mapping from key to route on download.pytorch.org/torchaudio. But the function itself does not have to use dict type for the mapping as long as the function serves as the correct mapping.

Reply:
sure, that works for now, but will probably need to be modified in the future (ex/ pretrained models for NNLM/lexicon free) where the file naming convention will be different

Reply:
> that would need to be modified in the future (ex/ when adding pretrained models for NNLM or lexicon free) where the naming convention will be different

Yeah, we can change it when we have the spec for the new additions. Until we have the spec (know how these new files will be placed) we need/should not take these in considerations. (this is called YAGNI principle. It is important because otherwise we sometimes cannot agree on how much of future expectation we want to put in the current implementation. And then priorities change and there are cases we end up giving up the said expected updates.)

Reply:
"librispeech-3/4-gram" key will load the pretrained files including the corresponding kenlm file, but there's also the option of "librispeech", which will load the lexicon and tokens but not an LM file, as there is no LM associated with just "librispeech". so in that case, we need to differentiate if we return a path for the lm path or `None`

Reply:
Isn't that logic already implemented at `lm=f"{prefix}/lm.bin" if model != "librispeech" else None`?

Reply:
yea that was added after the original comment -- just wanted to make sure hardcoding is fine for now since these are the only ones we are supporting now, sounds good

Reply:
Yup. that one LGTM. Please make sure to add validation on what keys are supported. that way the implementation contains sufficient information for the maintenance.